src/file-selector.ts

```diff
@@ -105,23 +105,29 @@ function flatten<T>(items: any[]): T[] {
   );
 }
 
-function fromDataTransferItem(
+async function fromDataTransferItem(
   item: DataTransferItem,
   entry?: FileSystemEntry | null,
-) {
-  if (typeof (item as any).getAsFileSystemHandle === "function") {
-    return (item as any).getAsFileSystemHandle().then(async (h: any) => {
-      const file = await h.getFile();
-      file.handle = h;
-      return toFileWithPath(file);
-    });
+): Promise<FileWithPath> {
+  if (typeof item.getAsFileSystemHandle === "function") {
+    const handle = await item.getAsFileSystemHandle();
+
+    if (!handle || !isFileHandle(handle)) {
+      throw new Error("Transferred item is not a file.");
```

Comment:
Should an error be thrown here? Throwing this error would still cause https://github.com/react-dropzone/file-selector/issues/120 to be reported, which is not desired.

Reply:
Not entirely sure, but we should already be [filtering out](https://github.com/react-dropzone/file-selector/blob/1a997bd800fcfa60d8523a69f4b361fee99965b9/src/file-selector.ts#L63) anything that is not a file. I am not really sure how we could get a handle back here, and it not be a file, but this is handling the edge cases we never expect to encounter.

Reply:
Gonna resolve this discussion, let's use your issue instead.